x/gamm/pool-models/balancer/pool.go

```diff
@@ -725,7 +725,7 @@ func (p *Pool) CalcJoinPoolShares(ctx sdk.Context, tokensIn sdk.Coins, swapFee s
 		tokensJoined = tokensIn
 		return numShares, tokensJoined, nil
 	} else if tokensIn.Len() != p.NumAssets() {
-		return sdk.ZeroInt(), sdk.NewCoins(), errors.New("balancer pool only supports LP'ing with one asset or all assets in pool")
+		return sdk.ZeroInt(), sdk.NewCoins(), types.ErrInvalidNumberOfCoins
```

Comment:
This actually causes a state-break. I would prefer that we keep this as a Go error please